### PR TITLE
Markdownlint: Add fixers for TOP005 and TOP006

### PR DESCRIPTION
**Pull request by MaoShizhong**
## Because
As of #27511, `markdownlint` is now officially a repo dependency as so fixers will be required for any custom rules that would be appropriate to have fixers for (i.e. not context-based).

## This PR
- Adds fix info for TOP005 and TOP006, so that any of those rule errors can be fixed when running the `fix:lesson` npm script.
- Rewrites TOP006 function in order to allow nested code blocks to be flagged and fixed as necessary. The previous token-based approach would not allow this due to the way markdown-it parses fences.
- Adds `sh` and `yml` to the abbreviations map in TOP006.
- Updates TOP006 docs and test files to account for the new languages, and nested code block behaviour.

## Issue
N/A

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [ ] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [ ] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: markdownlint/TOP006_fullFencedCodeLanguage/TOP006_fullFencedCodeLanguage.js [hunk 1]
```diff
@@ -2,7 +2,9 @@ const LANGUAGES_WITH_ABBREVIATIONS = new Map()
   .set("js", "javascript")
   .set("rb", "ruby")
   .set("txt", "text")
-  .set("md", "markdown");
+  .set("md", "markdown")
+  .set("sh", "bash")
+  .set("yml", "yaml");
```
Comment (thatblindgeye) [inline markdownlint/TOP006_fullFencedCodeLanguage/TOP006_fullFencedCodeLanguage.js:7]: Can we add bash and yaml to the test file as examples?

**Review by thatblindgeye — CHANGES_REQUESTED** [review]

**Comment by MaoShizhong** [comment]
Had to account for an edge case where a tag was surrounded by a whitespace-only line.
Originally, it would insert a blank line, then MD009 wouldn't fix the whitespace-only line leaving us with a blank line followed by a whitespace-only line after running the fixer.

Now, running the fixer allows MD009 to fix the whitespace-only line, leaving it blank.

i.e. 
Original would go from 
```markdown
<div>

Text

</div>
\s
Some other text
```
to
```markdown
<div>

Text

</div>

\s
Some other text
```
But now it accounts for whitespace-only lines, it fixes it to
```markdown
<div>

Text

</div>

Some other text
```

**Review by thatblindgeye — APPROVED** [review]
